### PR TITLE
✨ ES Modules (CAL-373 CAL-393 CAL-394)

### PR DESCRIPTION
**Pull request by benschwarz**
In this PR: 

- [x] CLI is now written as ES Modules
- [x] Add build step for producing commonjs entrypoints
- [x] New package script: `npm run generate-cli-md` generates `CLI_COMMANDS.md`, Single file CLI documentation (Markdown)
- [x] New package export: `cli-commands`, returns all command functions and cli options.
- [x] New package export: `cli-metadata`, returns all command documentation.
- [x] Ends support for node 12 (EOL in 15 days)
- [x] Discontinues packaged binary builds, releases only via npm.
- [x] `calibre test create` has a new flag `--waitForTest` (default: false).
- [x] Updates all `examples/nodejs` to ES modules
- [x] Turn ESLint on for build process.
- [x] Update README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,44 +1,109 @@
...
+
+## 🛠 Usage
+To see a full list of available commands, subcommands and options run:
+
+```bash
 $ calibre --help
+$ calibre <command> --help
+```
+
+or [see all commands](commands.md) in this repository.
```
Comment (thefoxis) [inline README.md:64]: @benschwarz here's the upcoming link to the commands markdown file, feel free to amend the name.

Comment (benschwarz) [reply README.md:64]: @thefoxis Thanks! I'll update once I've dropped in the markdown render

File: CLI_COMMANDS.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
+
```
Comment (benschwarz) [inline CLI_COMMANDS.md:1]: @thefoxis `CLI_COMMANDS` shows all commands and sub-commands, inclusive of their descriptions, options and defaults. If you can think of any improvements from this first cut I'd be happy to include them! On first look it seems like a good start 🥲

Comment (thefoxis) [reply CLI_COMMANDS.md:1]: @benschwarz I think you can push it as is for now. we can improve later if we want to. 👍🏻

Comment (benschwarz) [reply CLI_COMMANDS.md:1]: Aight then! Thanks for your help. I'll update the changelog, cleanup and get this release shipped 🛥️

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,208 +1,284 @@
-### 4.2.0 - 2022-01-21
+# 5.0.0 (2022-04-29)
```
Comment (thefoxis) [inline CHANGELOG.md:1]: @benschwarz I added the changes for `5.0.0` here, you might have to update the date if that doesn't end up being the release date.

i also formatted the entire file so it's consistent and categorises changes for easier reading. you might want to have a scan if everything is appropriately categorised. 😅

Comment (benschwarz) [reply CHANGELOG.md:1]: Tomorrow is the day!! 🕐

**Pull request merged**